### PR TITLE
feat(metadata): implement GET Units of Measure API

### LINKED ISSUE #3957: Implement Unit of Measure (UoM) ADR
# 🚀 Feature Request

### Relevant Package [**REQUIRED**]

 This feature request is for core-metadata

### Description [**REQUIRED**]
Implement the design in the ADR
https://docs.edgexfoundry.org/2.2/design/adr/core/0022-UoM/
`GET /api/v2/uom`
In addition, I suggest to add a GET API to retrieve the current UoM settings.  It will just load the UoMFile and return the content.
Also, use the `Accept` header to determine whether the API should return JSON or TOML format.
Ref: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept

Furthermore, the `Values` of UoM should be String array
`Values=["lbs", "ounces", "kilos", "grams"]`
not
`Values="lbs,ounces,kilos,grams"`

### PR DESCRIPTION
**Pull request by hahattan**
dependent on https://github.com/edgexfoundry/go-mod-core-contracts/pull/756

cbc81fb9b27178ac0d33edb17f9e140940e39377 updated `uom.toml` configuration structure in order to return similar content for json/toml response. 

fix #3957 

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [x] I have opened a PR for the related docs change (if not, why?) The API document is also updated in this PR.
  <link to docs PR>

## Testing Instructions

1. Run core-metadata from this branch
2. 
```shell
$ curl localhost:59881/api/v2/uom
{
    "apiVersion": "v2",
    "statusCode": 200,
    "uom": {
        "source": "reference to source for all UoM if not specified below",
        "units": {
            "temperature": {
                "source": "www.weather.com",
                "values": [
                    "C",
                    "F",
                    "K"
                ]
            },
            "weights": {
                "source": "www.usa.gov/federal-agencies/weights-and-measures-division",
                "values": [
                    "lbs",
                    "ounces",
                    "kilos",
                    "grams"
                ]
            }
        }
    }
}

$ curl -H "Accept: application/toml" localhost:59881/api/v2/uom
Source = "reference to source for all UoM if not specified below"

[Units]

  [Units.temperature]
    Source = "www.weather.com"
    Values = ["C", "F", "K"]

  [Units.weights]
    Source = "www.usa.gov/federal-agencies/weights-and-measures-division"
    Values = ["lbs", "ounces", "kilos", "grams"]
```

### CODE REVIEW

**Comment by lenny-intel** [comment]
@hahattan , I noticed the `I am not introducing a breaking change` isn't checked in the PR checklist above. I assume this was just missed and there are not breaking changes :-)

File: internal/core/metadata/controller/http/uom_test.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+			recorder := httptest.NewRecorder()
+			handler := http.HandlerFunc(controller.UnitsOfMeasure)
+			handler.ServeHTTP(recorder, req)
+
+			if testCase.Accept == common.ContentTypeJSON {
+				actualResponse := responses.UnitsOfMeasureResponse{}
+				err = json.Unmarshal(recorder.Body.Bytes(), &actualResponse)
+				require.NoError(t, err)
+
+				assert.Equal(t, http.StatusOK, recorder.Result().StatusCode, "HTTP status code not as expected")
```
Comment (lenny-intel) [inline internal/core/metadata/controller/http/uom_test.go:67]: This is common for both content types. Move out of if/else

File: internal/core/metadata/controller/http/uom_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+				assert.Equal(t, common.ApiVersion, actualResponse.ApiVersion, "Api Version not as expected")
+				assert.Equal(t, http.StatusOK, actualResponse.StatusCode, "BaseResponse status code not as expected")
+			} else {
+				actualResponse := uom.UnitsOfMeasureImpl{}
+				err = toml.Unmarshal(recorder.Body.Bytes(), &actualResponse)
+				require.NoError(t, err)
+
```
Comment (lenny-intel) [inline internal/core/metadata/controller/http/uom_test.go:74]: add assert to verify actual actualResponse  equals expected response

File: internal/core/metadata/controller/http/uom_test.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+			handler := http.HandlerFunc(controller.UnitsOfMeasure)
+			handler.ServeHTTP(recorder, req)
+
+			if testCase.Accept == common.ContentTypeJSON {
+				actualResponse := responses.UnitsOfMeasureResponse{}
+				err = json.Unmarshal(recorder.Body.Bytes(), &actualResponse)
+				require.NoError(t, err)
+
+				assert.Equal(t, http.StatusOK, recorder.Result().StatusCode, "HTTP status code not as expected")
+				assert.Equal(t, common.ApiVersion, actualResponse.ApiVersion, "Api Version not as expected")
```
Comment (lenny-intel) [inline internal/core/metadata/controller/http/uom_test.go:68]: add assert to verify actual actualResponse  equals expected response

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Comment by hahattan** [comment]
> @hahattan , I noticed the `I am not introducing a breaking change` isn't checked in the PR checklist above. I assume this was just missed and there are not breaking changes :-)

not sure whether updating the `uom.toml` is a breaking change so I left it unchecked

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Comment by cloudxxx8** [comment]
> > @hahattan , I noticed the `I am not introducing a breaking change` isn't checked in the PR checklist above. I assume this was just missed and there are not breaking changes :-)
> 
> not sure whether updating the `uom.toml` is a breaking change so I left it unchecked

it's not a breaking change, because this file didn't exist in the previous release

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**